### ISSUE TITLE
[Bug] NOT LIKE parsing exception

### ISSUE THREAD

**Issue by Wilson-BT** [apache/doris#29323]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

doris-2.0.3

### What's Wrong?

When I execute this not like SQL, the Query in the parsing plan looks like this, causing the final query to be abnormal.
![image](https://github.com/apache/doris/assets/40427110/bc961c9e-c168-4be4-8cb0-91fcf98b7fc9)

### What You Expected?

Bug Fix

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by WinkerDu** [comment]
Can you provide more info to reproduce this issue, like `create table` statement and `query` SQL

**Comment by yiguolei** [comment]
@Wilson-BT Need table schema and query sql information.

**Comment by Wilson-BT** [comment]
Table schema:
```
CREATE TABLE `organization` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT COMMENT '自增主键',
  `unit_id` int(11) NOT NULL COMMENT '组织ID',
  `mchr_id` int(11) NULL COMMENT 'MCHR组织对照编码',
  `unit_code` varchar(50) NULL COMMENT '组织编码',
  `unit_level_name` varchar(50) NULL COMMENT '组织级别名称',
  `unit_level_id` int(11) NULL COMMENT '组织级别ID',
  `parent_id` int(11) NULL COMMENT '组织父ID-取父组织的MCHRID',
  `name` varchar(100) NULL COMMENT '组织简称',
  `full_name` varchar(150) NULL COMMENT '组织名称',
  `org_status` tinyint(4) NULL COMMENT '组织状态',
  `complement` int(11) NULL COMMENT '编制人数',
  `actual_comp_count` int(11) NULL COMMENT '实际编制人数',
  `owner_employee_name` varchar(50) NULL COMMENT '组织负责人名称',
  `owner_employee_id` int(11) NULL COMMENT '组织负责人ID',
  `owner_employee_code` varchar(50) NULL COMMENT '组织负责人工号',
  `parent_id2` int(11) NULL COMMENT '专业上级组织ID-取专业上级组织的MCHRID',
  `parent_id3` int(11) NULL COMMENT '指导上级组织ID-取指导上级组织的MCHRID',
  `hr_emp_no` varchar(50) NULL COMMENT 'HR工号',
  `hr_emp_title` varchar(50) NULL COMMENT 'HR姓名',
  `hr_head_emp_no` varchar(50) NULL COMMENT 'HR负责人工号',
  `hr_head_emp_title` varchar(50) NULL COMMENT 'HR负责人姓名',
  `pline_emp_no` varchar(50) NULL COMMENT '专业线负责人工号',
  `pline_emp_title` varchar(50) NULL COMMENT '专业线负责人姓名',
  `incharge_emp_no` varchar(50) NULL COMMENT '业务分管领导工号',
  `incharge_emp_title` varchar(50) NULL COMMENT '业务分管领导姓名',
  `brand_head_emp_no` varchar(50) NULL COMMENT '零售负责人工号',
  `brand_head_emp_title` varchar(50) NULL COMMENT '零售负责人姓名',
  `sort` int(11) NULL COMMENT '排序',
  `unit_type_id` int(11) NULL COMMENT '组织类型ID',
  `unit_type_name` varchar(50) NULL COMMENT '组织类型名称',
  `cost_center_id` int(11) NULL COMMENT '成本中心ID',
  `cost_center` varchar(50) NULL COMMENT '成本中心',
  `functional_property_id` int(11) NULL COMMENT '职能性质ID',
  `functional_property` varchar(20) NULL COMMENT '职能性质',
  `city_no` varchar(50) NULL COMMENT '城市编号',
  `location_city` varchar(50) NULL COMMENT '所在城市',
  `zone_no` varchar(50) NULL COMMENT '地区编号',
  `zone_name` varchar(50) NULL COMMENT '地区',
  `province_no` varchar(50) NULL COMMENT '省区编号',
  `province_name` varchar(50) NULL COMMENT '省区',
  `manager_city_no` varchar(50) NULL COMMENT '管理城市编号',
  `manger_city` varchar(50) NULL COMMENT '管理城市',
  `biz_city_no` varchar(50) NULL COMMENT '经营城市编号',
  `biz_city` varchar(50) NULL COMMENT '经营城市',
  `is_center_city` varchar(20) NULL COMMENT '是否中心城市',
  `store_code` varchar(50) NULL COMMENT '店铺编码',
  `store_name` varchar(50) NULL COMMENT '店铺名称',
  `brand` varchar(50) NULL COMMENT '主品牌',
  `other_brand` varchar(100) NULL COMMENT '其他品牌',
  `company_no` varchar(50) NULL COMMENT '公司编号',
  `company_name` varchar(100) NULL COMMENT '公司名称',
  `address` varchar(200) NULL COMMENT '详细地址',
  `shop_no` varchar(50) NULL COMMENT '零售代码',
  `store_level` varchar(50) NULL COMMENT '店铺级别',
  `shop_type` varchar(50) NULL COMMENT '店铺类别',
  `shop_date` varchar(20) NULL COMMENT '开启日期',
  `drawal_date` varchar(20) NULL COMMENT '撤柜日期',
  `market_name` varchar(100) NULL COMMENT '商场名称',
  `store_nature` varchar(50) NULL COMMENT '店铺性质',
  `man_emp_no` varchar(50) NULL COMMENT '多门店负责人工号',
  `man_emp_title` varchar(50) NULL COMMENT '多门店负责人姓名',
  `create_date` varchar(20) NULL COMMENT 'EHR创建日期',
  `disabled_date` varchar(20) NULL COMMENT 'EHR失效日期',
  `identification` varchar(50) NULL COMMENT '品牌标识',
  `bigstore` tinyint(4) NULL COMMENT '是否大店',
  `orgtpath` varchar(100) NULL COMMENT '组织路径',
  `data_status` tinyint(4) NULL COMMENT '数据状态: 1-新增|2-修改|3-失效',
  `uptdate` varchar(20) NULL COMMENT 'EHR更新日期',
  `supervisor_emp_no` varchar(50) NULL COMMENT '门店陈列师工号',
  `supervisor_title` varchar(50) NULL COMMENT '门店陈列师姓名',
  `director_emp_no` varchar(50) NULL COMMENT '陈列专家工号',
  `director_title` varchar(50) NULL COMMENT '陈列专家姓名',
  `trainer_emp_no` varchar(50) NULL COMMENT '培训主管工号',
  `trainer_title` varchar(50) NULL COMMENT '培训主管姓名',
  `training_emp_no` varchar(50) NULL COMMENT '培训经理工号',
  `training_title` varchar(50) NULL COMMENT '培训经理姓名',
  `operator_emp_no` varchar(50) NULL COMMENT '线上运营主管工号',
  `operator_title` varchar(50) NULL COMMENT '线上运营主管姓名',
  `operating_emp_no` varchar(50) NULL COMMENT '线上运营经理工号',
  `operating_title` varchar(50) NULL COMMENT '线上运营经理姓名',
  `area_no` varchar(50) NULL COMMENT '大区编号',
  `area_name` varchar(50) NULL COMMENT '大区',
  `update_time` datetimev2(0) NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '更新时间',
  `create_time` datetimev2(0) NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间'
) ENGINE=JDBC_EXTERNAL_TABLE;
```
Sql ：
```
  select *
  from tidb_prod_tikv.ehr_sync.organization 
 where (full_name not like '%虚店%' )and (full_name not like '%直播%')
```

**Comment by Wilson-BT** [comment]
> @Wilson-BT Need table schema and query sql information.

it`s a jdbc catalog table, table schema:
```
CREATE TABLE `organization` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT COMMENT '自增主键',
  `full_name` varchar(150) NULL COMMENT '组织名称',
  `update_time` datetimev2(0) NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '更新时间',
  `create_time` datetimev2(0) NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间'
) ENGINE=JDBC_EXTERNAL_TABLE;
```
Sql ：
```
  select *
  from tidb_prod_tikv.ehr_sync.organization 
 where (full_name not like '%虚店%' )and (full_name not like '%直播%')
```

I think,you can create any jdbc catalog table, and use `not like 'xxx'` to reproduce this issue.

**Comment by WinkerDu** [comment]
I try to create jdbc catalog for external mysql table:
```
CREATE TABLE `organization` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT PRIMARY KEY,
  `full_name` varchar(150) NULL,
  `update_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
)
```
then execute query like:
```
explain select *
from mysql_jdbc_catalog.drp_db.organization 
 where (full_name not like '%虚店%' )and (full_name not like '%直播%');
```
I cannot reproduce this issue in `Branch-2.0` branch for both Nereids planner and Old planner.
Nereids planner:
<img width="1727" alt="image" src="https://github.com/apache/doris/assets/11454734/704cb065-a956-4cf8-9c0a-afbafce5cda7">
Old planner:
<img width="1727" alt="image" src="https://github.com/apache/doris/assets/11454734/7baae90b-d6dd-437d-bbcf-7a286a998fff">
Maybe you can upgrade your Doris? @Wilson-BT 
cc @yiguolei

**Comment by Wilson-BT** [comment]
> I try to create jdbc catalog for external mysql table:
> 
> ```
> CREATE TABLE `organization` (
>   `id` bigint(20) NOT NULL AUTO_INCREMENT PRIMARY KEY,
>   `full_name` varchar(150) NULL,
>   `update_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
>   `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
> )
> ```
> 
> then execute query like:
> 
> ```
> explain select *
> from mysql_jdbc_catalog.drp_db.organization 
>  where (full_name not like '%虚店%' )and (full_name not like '%直播%');
> ```
> 
> I cannot reproduce this issue in `Branch-2.0` branch for both Nereids planner and Old planner. Nereids planner: <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/11454734/293579623-704cb065-a956-4cf8-9c0a-afbafce5cda7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxNjIzODMsIm5iZiI6MTcwNDE2MjA4MywicGF0aCI6Ii8xMTQ1NDczNC8yOTM1Nzk2MjMtNzA0Y2IwNjUtYTk1Ni00Y2Y4LTljMGEtYWZiYWZjZTVjZGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDAyMjEyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhOGU1YmIxY2MyZTdiZmUwYjczNjUwZjhkZmQyYjZlZjg0MjI5YzBlMWQwOWU0YWZlMTQ5YzdhMjUxNzhjMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TWuot_-qkDK-MAMfxITNXAZDWHRYJuLS0_utK2-x9hs"> Old planner: <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/11454734/293579708-7baae90b-d6dd-437d-bbcf-7a286a998fff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxNjIzODMsIm5iZiI6MTcwNDE2MjA4MywicGF0aCI6Ii8xMTQ1NDczNC8yOTM1Nzk3MDgtN2JhYWU5MGItZDZkZC00MzdkLWJiY2YtN2EyODZhOTk4ZmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDAyMjEyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmZDVhNTkzYjZjOWEzZGQ2NDU3MDZlOGVjNjJhNmM2ZDA0Y2I3MTRhODg0ZTM0ZDA2OTFlOGZmMGEwNjZiZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uabZepokmwOogiPVVizZdXWno3vP68HG3BgGweLQk20"> Maybe you can upgrade your Doris? @Wilson-BT cc @yiguolei

OK, thanks, i will upgrade now

**Comment by baofengzhouyu** [comment]
> I try to create jdbc catalog for external mysql table:
> 
> ```
> CREATE TABLE `organization` (
>   `id` bigint(20) NOT NULL AUTO_INCREMENT PRIMARY KEY,
>   `full_name` varchar(150) NULL,
>   `update_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
>   `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
> )
> ```
> 
> then execute query like:
> 
> ```
> explain select *
> from mysql_jdbc_catalog.drp_db.organization 
>  where (full_name not like '%虚店%' )and (full_name not like '%直播%');
> ```
> 
> I cannot reproduce this issue in `Branch-2.0` branch for both Nereids planner and Old planner. Nereids planner: <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/11454734/293579623-704cb065-a956-4cf8-9c0a-afbafce5cda7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNTY3MDAsIm5iZiI6MTcwNTA1NjQwMCwicGF0aCI6Ii8xMTQ1NDczNC8yOTM1Nzk2MjMtNzA0Y2IwNjUtYTk1Ni00Y2Y4LTljMGEtYWZiYWZjZTVjZGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDEwNDY0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZjYTcyZTM1YzIxMTFlOTYwY2E3M2ZmN2JhYWYwZGE0OGVjMzI5NTFmNTcxOTBiNTVkNzQ3ODMxMzIwMGIwYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hjVk8p5-ZYHFnh52vx4QCcqJZRnhKygIg61VSE4iPOM"> Old planner: <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/11454734/293579708-7baae90b-d6dd-437d-bbcf-7a286a998fff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNTY3MDAsIm5iZiI6MTcwNTA1NjQwMCwicGF0aCI6Ii8xMTQ1NDczNC8yOTM1Nzk3MDgtN2JhYWU5MGItZDZkZC00MzdkLWJiY2YtN2EyODZhOTk4ZmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDEwNDY0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjYTk5YjE4YjZkZDI0Y2VhNWU4ZTZkNGNiZDgxMmU3ZGEwODBmNjZlNDJlNDExMzBjZDkxM2UzMDUwOTgwNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6V9n2R62Uz3RaYrp-EqOmlGiyyjFHQWuGJu0-R2mknY"> Maybe you can upgrade your Doris? @Wilson-BT cc @yiguolei

I have the same problem.  
version: 2.0.3
Is there a bug in the latest version ?

**Comment by Wilson-BT** [comment]
> > I try to create jdbc catalog for external mysql table:
> > ```
> > CREATE TABLE `organization` (
> >   `id` bigint(20) NOT NULL AUTO_INCREMENT PRIMARY KEY,
> >   `full_name` varchar(150) NULL,
> >   `update_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
> >   `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
> > )
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > then execute query like:
> > ```
> > explain select *
> > from mysql_jdbc_catalog.drp_db.organization 
> >  where (full_name not like '%虚店%' )and (full_name not like '%直播%');
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I cannot reproduce this issue in `Branch-2.0` branch for both Nereids planner and Old planner. Nereids planner: <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/11454734/293579623-704cb065-a956-4cf8-9c0a-afbafce5cda7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNTY3MDAsIm5iZiI6MTcwNTA1NjQwMCwicGF0aCI6Ii8xMTQ1NDczNC8yOTM1Nzk2MjMtNzA0Y2IwNjUtYTk1Ni00Y2Y4LTljMGEtYWZiYWZjZTVjZGE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDEwNDY0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZjYTcyZTM1YzIxMTFlOTYwY2E3M2ZmN2JhYWYwZGE0OGVjMzI5NTFmNTcxOTBiNTVkNzQ3ODMxMzIwMGIwYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hjVk8p5-ZYHFnh52vx4QCcqJZRnhKygIg61VSE4iPOM"> Old planner: <img alt="image" width="1727" src="https://private-user-images.githubusercontent.com/11454734/293579708-7baae90b-d6dd-437d-bbcf-7a286a998fff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNTY3MDAsIm5iZiI6MTcwNTA1NjQwMCwicGF0aCI6Ii8xMTQ1NDczNC8yOTM1Nzk3MDgtN2JhYWU5MGItZDZkZC00MzdkLWJiY2YtN2EyODZhOTk4ZmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDEwNDY0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjYTk5YjE4YjZkZDI0Y2VhNWU4ZTZkNGNiZDgxMmU3ZGEwODBmNjZlNDJlNDExMzBjZDkxM2UzMDUwOTgwNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6V9n2R62Uz3RaYrp-EqOmlGiyyjFHQWuGJu0-R2mknY"> Maybe you can upgrade your Doris? @Wilson-BT cc @yiguolei
> 
> I have the same problem. version: 2.0.3 Is there a bug in the latest version ?

It may have been fixed in 2.0.4. Just wait and see
